slack_today_i_did/reports.py

```diff
@@ -110,3 +110,61 @@ def as_dict(self):
             'people_to_bother': self.people_to_bother,
             'time_run': (str(self.time_run) if self.time_run is not None else "")  # noqa: E501
         }
+
+
+class Sessions(object):
+    def __init__(self):
+        self.sessions = {}
+
+    def has_running_session(self, person: str) -> bool:
+        if person not in self.sessions:
+            return False
+
+        return self.sessions[person]['is_running']
+
+    def start_session(self, person: str, channel: str) -> None:
+        self.sessions[person] = {
+            'is_running': True,
+            'messages': [],
+            'channel': channel
+        }
+
+    def end_session(self, person: str) -> None:
+        if not self.has_running_session(person):
+            return
+
+        self.sessions[person]['is_running'] = False
+
+    def add_message(self, person: str, message: str) -> None:
+        if not self.has_running_session(person):
+            return
+
+        self.sessions[person]['messages'].append(message)
+
+    def get_entry(self, person: str) -> Dict[str, Any]:
+        if person not in self.sessions:
+            return {}
+
+        return self.sessions[person]
+
+    def retire_session(self, person: str, filename: str) -> None:
+        session_info = self.sessions.pop(person)
+
+        with open(filename, 'w') as f:
+            json.dump(session_info, f)
+
+    def load_from_file(self, filename: str) -> None:
+        """ Load people:session from a file """
+        try:
+            with open(filename) as f:
+                as_json = json.load(f)
+        except FileNotFoundError:
+            return
+
+        for (name, session) in as_json['sessions'].items():
+            self.sessions[name] = session
```

Comment:
❔ is this to preserve session keys that are not in the file? should that behavior be tested?